snapcraft_legacy/plugins/v2/_kernel_build.py

```diff
@@ -1111,7 +1111,7 @@ def _arrange_install_dir_cmd(install_dir: str) -> List[str]:
             # but snapd expects modules/ and firmware/
             mv {install_dir}/lib/modules {install_dir}/
             # remove symlinks modules/*/build and modules/*/source
-            rm {install_dir}/modules/*/build {install_dir}/modules/*/source
+            rm -rf {install_dir}/modules/*/build {install_dir}/modules/*/source
```

Comment:
If that's the symlink, I'd avoid adding `-r` as it seems unnecessary in that case:

```suggestion
            rm -f {install_dir}/modules/*/build {install_dir}/modules/*/source
```

Reply:
hi @dariuszd21 , 

thanks for the feedback! previously I was considering adding `-f` only, the reason is same as your opinion, but I noticed that `-rf` was used in the same file when using `rm`, such as

https://github.com/canonical/snapcraft/blob/a9badaa32be6bcfa24f99302a3d6dffecd96063b/snapcraft_legacy/plugins/v2/_kernel_build.py#L739

and

https://github.com/canonical/snapcraft/blob/a9badaa32be6bcfa24f99302a3d6dffecd96063b/snapcraft_legacy/plugins/v2/_kernel_build.py#L1048

The above cases are obviously removing a file instead of a directory, so using `-r` seems meaningless. But in order to make sure the consistency in that file and prevent any regression(though looks like there won't be), I changed my mind to use `-rf`.

Or do you want me to remove all `-r` for those are actually not removing a directory?

Reply:
`-rf` raise a flag for me :D
I don't think it's necessary to remove it everywhere else as a part of this PR, but to keep it consistent and compatible with previous implementation I'd just stick with `-f` (or use `-e` guard, to skip removing if it does not exist).

Reply:
In that case if it will be unexpectedly the directory it will fail, instead of removing it silently :smile:

Reply:
Got it, I just force-pushed the changes, also updated the `tests/unit/parts/plugins/test_kernel.py`  as @mr-cal has requested, please let me know if you have any concerns, thanks!